db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj

```diff
@@ -2359,7 +2359,7 @@ public class SQLParser
             if (inSection && part.equalsIgnoreCase("with")) {
                 inSection = false;
             } else if (inSection) {
-                buf.append(part).append(" ");
+                buf.append(part).append("\n");
```

Comment:
If we have multiple words in the comment, this fix may not work. For example:
```
create table t2 select a1 from t1 --comment where b1>3
where b1 < 3 with data;
```
We could be putting "where b1>3" in the comment in a separate line, and this would cause syntax error.

Reply:
Thanks Yi... I think the pattern matching for the WITH clause needs to be fixed here too because it could be matching the word "with" inside a comment string.

Reply:
@yxia92 I reworked the fixed to explicitly match C-style comments and  '--' comment strings, except for --splice-properties hints, using regular expressions.  Please take a look.